packages/render/src/primitives/renderNode.js

```diff
@@ -12,8 +12,18 @@ import setLink from '../operations/setLink';
 import clipNode from '../operations/clipNode';
 import transform from '../operations/transform';
 import setDestination from '../operations/setDestination';
-
-const isRecursiveNode = (node) => node.type !== P.Text && node.type !== P.Svg;
+import renderForm from './form/renderForm';
+import renderFormField from './form/renderFormField';
+import renderTextInput from './form/renderTextInput';
+import renderFormPushButton from './form/renderFormPushButton';
+import renderPicker from './form/renderPicker';
+import renderFormList from './form/renderFormList';
+
+const isRecursiveNode = (node) =>
+  node.type !== P.Text &&
+  node.type !== P.Svg &&
+  node.type !== P.Form &&
+  node.type !== P.FormField;
```

Comment:
I ran into the following error when at least `node.type !== P.FormField`  is not applied: https://github.com/traveltechdeluxe/react-pdf/blob/81b75d26cf57e307256147372718ab8246676af6/packages/render/src/primitives/form/renderTextInput.js#L9-L12

Reply:
This could be solved by introducing a cleanUp function similar to the render function in the renderNode.js, which would be called after render the component and childs. This way a form field would not need to render its own childs. 

```suggestion
const isRecursiveNode = (node) =>
  node.type !== P.Text &&
  node.type !== P.Svg &&
  node.type !== P.Form;

const cleanUpFns = {
  [P.FormField]: cleanUpFormField
}

// inside renderNode():

  const renderFn = renderFns[node.type];

  if (renderFn) renderFn(ctx, node, options);

  if (shouldRenderChildren) renderChildren(ctx, node, options);

  const cleanUpFn = cleanUpFns[node.type];

  if (cleanUpFn) cleanUpFn(ctx, node, options);

// renderFormField.js:

const renderFormField = (ctx, node, options) => {
  const name = node.props?.name || '';
  
  const formField = ctx.formField(name);
  const option = options;
  if(!option.formFields)
    option.formFields = [formField]
  else
  option.formFields.push(formField)
};

const cleanUpFormField = (_ctx, _node, options) => {
  options.formFields.pop()
}
```

Reply:
On the form text it would then need something like this: `options.formFields?.at(0)`